src/rules/no-trailing-spaces.js

```diff
@@ -6,7 +6,7 @@ function noTrailingSpaces(parsedFile, fileName) {
   var lines = fs.readFileSync(fileName).toString().split('\n');
   var lineNo = 1;
   lines.forEach(function(line) {
-    if (line[line.length - 1] == ' ') {
+    if (/[\s]+$/.test(line)) {
```

Comment:
I think the regex approach is good. Trim also removes whitespace from the beginning of a string, so i don't think we could simply do `if (line != line.trim())`